File: Problem_Solving/Algorithms/Implementation/1_Grading_Students/Solution.php
<?php

function gradingStudents($grades) {
    $minimumGrade=38;
    $gradesLen=count($grades);
    $roundedGrades=array();

    for($i=0;$i<$gradesLen;$i++){
        //round if distance<=2 from ith grade to next multiple of 5
        if($grades[$i]>=$minimumGrade && $grades[$i]%5>=3)
            $roundedGrades[]=($grades[$i]-($grades[$i]%5)+5);
        else
            $roundedGrades[]=$grades[$i];
    }

    return $roundedGrades;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$grades_count = intval(trim(fgets(STDIN)));

$grades = array();

for ($i = 0; $i < $grades_count; $i++) {
    $grades_item = intval(trim(fgets(STDIN)));
    $grades[] = $grades_item;
}

$result = gradingStudents($grades);

fwrite($fptr, implode("\n", $result));

fclose($fptr);

?>

File: Problem_Solving/Algorithms/Implementation/1_Grading_Students/Solution.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result {

    public static List<int> gradingStudents(List<int> grades){
        int MINIMUM_GRADE=38;
        List<int> roundedGrades=new List<int>(grades.Count);

        foreach(int grade in grades){
            //round if distance<=2 from ith grade to next multiple of 5
            if(grade>=MINIMUM_GRADE && grade%5>=3)
                roundedGrades.Add(grade-(grade%5)+5);
            else
                roundedGrades.Add(grade);
        }

        return roundedGrades;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int gradesCount = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> grades = new List<int>();

        for (int i = 0; i < gradesCount; i++)
        {
            int gradesItem = Convert.ToInt32(Console.ReadLine().Trim());
            grades.Add(gradesItem);
        }

        List<int> result = Result.gradingStudents(grades);

        textWriter.WriteLine(String.Join("\n", result));

        textWriter.Flush();
        textWriter.Close();
    }
}
